### PR TITLE
Refactor/save artifacts in db

### PR DESCRIPTION
**Pull request by jakebeckstrom**
Refactor api to use process_ticket to handle tickets and save the returned artifacts in the database

### CODE REVIEW

File: models/ticket.go [hunk 1]
```diff
@@ -12,10 +20,80 @@ type Ticket struct {
 	ScreenShot []ScreenShot `json:"screenshots"`
 }
 
+type HoneyclientRequest struct {
```
Comment (williamtdr) [inline models/ticket.go]: This doesn’t need to be its own type, we can just pass the ticket object as is

File: models/ticket.go [hunk 2]
```diff
@@ -12,10 +20,80 @@ type Ticket struct {
...
+		// Get file artifact from in memory storage
+		resp, err := http.Get("http://localhost:8000" + s)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		defer resp.Body.Close()
+
+		// Decode actual file artifact into our struct, set other fields
+		resData, err := ioutil.ReadAll(resp.Body)
```
Comment (williamtdr) [inline models/ticket.go:61]: Nice use of `ioutil.ReadAll` :)

**Review by williamtdr — CHANGES_REQUESTED** [review]

File: models/ticket.go (hunk 1 above)
Comment (rafibarash) [reply models/ticket.go]: just updated this

**Review by rafibarash — APPROVED** [review]
Tested this out with the frontend and honeyclient master branches, working well together. I'm going to go ahead and merge this so we can work off all the master branches again

**Review by williamtdr — APPROVED** [review]
lgtm!

**Pull request merged**